src/commands/shared.ts

```diff
@@ -187,10 +187,19 @@ export const provisionRequest = async (
     throw "Public key mismatch. Please revoke the request and try again.";
   }
 
-  return { request: provisionedRequest, publicKey, privateKey };
+  const posixAccount =
+    provisionedRequest.permission.spec.type === "gcloud"
+      ? await importSshKey(publicKey, { debug: args.debug })
+      : undefined;
+  const linuxUserName = posixAccount?.username;
+
+  return { request: provisionedRequest, publicKey, privateKey, linuxUserName };
 };
 
-export const requestToSsh = (request: Request<PluginRequest>): SshRequest => {
+export const requestToSsh = (
+  request: Request<PluginRequest>,
+  options: { gcloud: { linuxUserName?: string } }
```

Comment:
Isn't the `linuxUserName` already on the request object? Why not just dither on the request type directly?

Reply:
For GCP requests, we are not using the request object's `linuxUserName`. Actually it's not even in the request object for GCP.